enclave/sgx/eeid_attester.c

```diff
@@ -0,0 +1,178 @@
+
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef OE_BUILD_ENCLAVE
+#include <openenclave/enclave.h>
+#else
+#include <openenclave/host.h>
+#endif
+
+#include <openenclave/attestation/sgx/eeid_attester.h>
+#include <openenclave/bits/eeid.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/sgx/eeid_plugin.h>
+#include <openenclave/internal/sgx/plugin.h>
+#include <openenclave/internal/trace.h>
+
+extern const void* __oe_get_eeid();
+
+#ifdef OE_BUILD_ENCLAVE
+#include <openenclave/attestation/sgx/attester.h>
+#include <openenclave/enclave.h>
+#else
+#include <openenclave/host.h>
+#endif
+
+oe_result_t eeid_attester_on_register(
+    oe_attestation_role_t* context,
+    const void* config_data,
+    size_t config_data_size)
+{
+    OE_UNUSED(context);
+    OE_UNUSED(config_data);
+    OE_UNUSED(config_data_size);
+
+    oe_result_t result = OE_UNEXPECTED;
+
+#ifdef OE_BUILD_ENCLAVE
+    oe_attester_t* sgx_attest = oe_sgx_plugin_attester();
+    result = oe_register_attester(sgx_attest, NULL, 0);
+    if (result != OE_ALREADY_EXISTS)
+        return result;
+#else
+    return OE_UNSUPPORTED;
+#endif
+
+    return result;
+}
+
+oe_result_t eeid_attester_on_unregister(oe_attestation_role_t* context)
+{
+    OE_UNUSED(context);
+    oe_result_t result = OE_UNEXPECTED;
+
+#ifdef OE_BUILD_ENCLAVE
```

Comment:
Isn't OE_BUILD_ENCLAVE always defined for this file?

Reply:
Yes, in this version it is. Fixed.